tests/voq/test_voq_fabric_capacity.py

```diff
@@ -0,0 +1,139 @@
+
+logger = logging.getLogger(__name__)
+
+# This test only runs on t2 systems.
+pytestmark = [
+    pytest.mark.topology('t2')
+]
+
+# This test checks the output of the "show fabric monitor capacity" command
+# on a linecard. It is designed to run on a modular chassis.
+
+
+def test_fabric_capacity(duthosts, enum_frontend_dut_hostname):
+    """Checks if the fabric capacity monitor works"""
+
+    # get a start state from system
+    # by running "show fabric monitor capcity" command
+    duthost = duthosts[enum_frontend_dut_hostname]
+    asic = 0
+    if duthost.is_multi_asic:
+        asic = random.randint(0, duthost.num_asics())
+    asicName = "asic{}".format(asic)
+    logger.info(asicName)
+
+    if duthost.is_multi_asic:
+        cmd = "show fabric monitor capacity -n asic{}".format(asic)
+    else:
+        cmd = "show fabric monitor capacity"
+    # The output of this show command is:
+    # # show fabric monitor capacity
+    # Monitored fabric capacity threshold:  100%
+    #
+    #   ASIC    Operating     Total #    %    Last Event    Last Time
+    #               Links    of Links
+    # ------  -----------  ----------  ---  ------------  -----------
+    #  asic0          112         112  100          None        Never
+    cmd_output = duthost.shell(cmd, module_ignore_errors=True)["stdout"].split("\n")
+    operating_links = 0
+    for line in cmd_output:
+        if not line:
+            continue
+        token = line.split()
+        if token[0].startswith("asic"):
+            operating_links = int(token[1])
+
+    # get list of up/unisolated links by running "show fabric isolation" with Isolated=0
+    # example output:
+    # # show fabric isolation -n asic0
+    #
+    # asic0
+    #   Local Link    Auto Isolated    Manual Isolated    Isolated
+    # ------------  ---------------  -----------------  ----------
+    #           20                0                  0           0
+    #           21                0                  0           0
+    up_link_list = []
+    if duthost.is_multi_asic:
+        cmd = "show fabric isolation -n asic{}".format(asic)
+    else:
+        cmd = "show fabric isolation"
+
+    cmd_output = duthost.shell(cmd, module_ignore_errors=True)["stdout"].split("\n")
+    for line in cmd_output:
+        if not line:
+            continue
+        token = line.split()
+        if not token[0].isdigit():
+            continue
+        localPort = token[0]
+        isolateSt = token[3]
+        if isolateSt == "0":
+            up_link_list.append(localPort)
+
+    if len(up_link_list) > 0:
+        shutlink = random.choice(up_link_list)
+    else:
+        return
+
+    # Start the test. Isolate a link and check if the capacity command get updated.
+    # Unisolate the link and check if the capacity command get updated.
+    try:
+        if duthost.is_multi_asic:
+            cmdPrefix = "sonic-db-cli -n asic{} APPL_DB hset 'FABRIC_PORT_TABLE:Fabric{}'".format(asic, shutlink)
```

Comment:
can we use the CLI `sudo config fabric port isolate` to isolate a fabric port

Reply:
> can we use the CLI `sudo config fabric port isolate` to isolate a fabric port

revised